cardano-testnet/src/Testnet/Util/Cli.hs

```diff
@@ -0,0 +1,143 @@
+module Testnet.Util.Cli
+  ( cliAddressKeyGen
+  , cliNodeKeyGen
+  , cliNodeKeyGenVrf
+  , cliNodeKeyGenKes
+  , cliStakeAddressKeyGen
+  , KeyNames (..)
+
+  , File (..)
+  , Address
+  , Operator -- == Cold Keys
+  , Kes
+  , StakeAddress
+  , Vrf
+
+  , VKey
+  , SKey
+
+  , OperatorCounter
+  , ByronKey
+  , ByronAddress
+
+  , ByronDelegationKey
+  , ByronDelegationCert
+
+  , getVKeyPath
+  , getSKeyPath
+
+  , cliKeyGen
+
+  , cliSigningKeyAddress
+  ) where
+
+import           Prelude
+
+import           System.FilePath.Posix
+
+import qualified Hedgehog.Extras.Test.Base as H
+import qualified Hedgehog.Extras.Test.File as H (writeFile)
+
+import           Testnet.Util.Process
+
+data KeyNames = KeyNames
+  { verificationKeyFile :: FilePath
+  , signingKeyFile :: FilePath
+  }
+
+type KeyGen a = H.Integration (File (a VKey), File (a SKey))
+
+cliAddressKeyGen :: TmpDir -> KeyNames -> KeyGen Address
+cliAddressKeyGen = shelleyKeyGen "address" "key-gen"
+
+cliStakeAddressKeyGen :: TmpDir -> KeyNames -> KeyGen StakeAddress
+cliStakeAddressKeyGen = shelleyKeyGen "stake-address" "key-gen"
+
+cliNodeKeyGenVrf :: TmpDir -> KeyNames -> KeyGen Vrf
+cliNodeKeyGenVrf = shelleyKeyGen "node" "key-gen-VRF"
+
+cliNodeKeyGenKes :: TmpDir -> KeyNames -> KeyGen Kes
+cliNodeKeyGenKes = shelleyKeyGen "node" "key-gen-KES"
+
+shelleyKeyGen :: String -> String -> TmpDir -> KeyNames -> KeyGen x
+shelleyKeyGen command subCommand tmpDir keyNames = do
+  let
+    vKeyPath = tmpDir </> verificationKeyFile keyNames
+    sKeyPath = tmpDir </> signingKeyFile keyNames
+  execCli_
+      [ command, subCommand
+      , "--verification-key-file", vKeyPath
+      , "--signing-key-file", sKeyPath
+      ]
+  return (File vKeyPath, File sKeyPath)
+
+cliNodeKeyGen
+  :: TmpDir
+  -> FilePath
+  -> FilePath
+  -> FilePath
+  -> H.Integration (File (Operator VKey), File (Operator SKey), File OperatorCounter)
+cliNodeKeyGen tmpDir vkey skey counter = do
+  let
+    vkPath = tmpDir </> vkey
+    skPath = tmpDir </> skey
+    counterPath = tmpDir </> counter
+  execCli_
+    [ "node", "key-gen"
+    , "--cold-verification-key-file", vkPath
+    , "--cold-signing-key-file", skPath
+    , "--operational-certificate-issue-counter-file", counterPath
+    ]
+  return (File vkPath, File skPath, File counterPath)
+
+data Address key
+data Kes key
+data Operator key
+data StakeAddress key
+data Vrf key
+
+data VKey
+data SKey
+
+data OperatorCounter
+data ByronKey
+data ByronAddress
+data ByronDelegationKey
+data ByronDelegationCert
```

Comment:
Haddocks for the above types would be very useful as HLS would show them in popups.  Particularly for `Operator`.